src/TheMovie/TheMovie/ViewModels/UpcomingMovieViewModel.cs

```diff
@@ -0,0 +1,198 @@
+
+        private int _pageUpcomingMovies = 0;
+        private int _totalPagesUpcomingMovies = 0;
+        private bool _isSearchUpcomingMovies = false;
+
+        private bool _isUpcomingMovieRefresh;
+        public bool IsUpcomingMovieRefresh
+        {
+            get => _isUpcomingMovieRefresh;
+            set => SetProperty(ref _isUpcomingMovieRefresh, value);
+        }
+
+        private bool _isLoading;
+        public bool IsLoading
+        {
+            get => _isLoading;
+            set => SetProperty(ref _isLoading, value);
+        }
+
+        private bool _isLoadingUpcomingMovies;
+        public bool IsLoadingUpcomingMovies
+        {
+            get => _isLoadingUpcomingMovies;
+            set => SetProperty(ref _isLoadingUpcomingMovies, value);
+        }
+
+        private string _searchPlaceHolder;
+        public string SearchPlaceHolder
+        {
+            get => _searchPlaceHolder;
+            set => SetProperty(ref _searchPlaceHolder, value);
+        }
+
+        private string _searchText;
+        public string SearchText
+        {
+            get => _searchText;
+            set => SetProperty(ref _searchText, value);
+        }
+
+        public ObservableCollection<UpcomingMovieModel> UpcomingMovies { get; set; }
+
+        #endregion
+
+        #region Commands
+
+        public Command SearchCommand { get; set; }
+        public Command UpcomingMovieRefreshCommand { get; set; }
+        public DelegateCommand<UpcomingMovieModel> ItemAppearingCommand { get; set; }
+        public DelegateCommand<UpcomingMovieModel> ItemTappedCommand { get; set; }
+
+        #endregion
+
+        #region Services
+
+        private readonly IMovieService _movieService;
+        private readonly IGenreService _genreService;
+
+        #endregion
+
+        public UpcomingMovieViewModel(INavigationService navigationService, IPageDialogService pageDialogService, IMovieService movieService, IGenreService genreService)
+            : base(navigationService, pageDialogService)
+        {
+            _movieService = movieService;
+            _genreService = genreService;
+            Title = "Upcoming Movies";
+            SearchPlaceHolder = "Search your movie :)";
+            UpcomingMovies = new ObservableCollection<UpcomingMovieModel>();
+            SearchCommand = new Command(async () => await ExecuteSearchCommand());
+            UpcomingMovieRefreshCommand = new Command(async () => await ExecuteUpcomingMovieRefreshCommand());
+            ItemAppearingCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemAppearingCommand);
+            ItemTappedCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemTappedCommand);
+            InitializeUpcomingMovies();
+        }
+
+        #region Methods 
+
+        private async Task GetUpcomingMovies(int pageMovie = 1)
+        {
+            var movies = await _movieService.GetAll(pageMovie);
+
+            if (movies.UpcomingMovies.Count() > 0)
```

Comment:
How the execution would differ here if you used `Length > 0` or `Any()` instead of `Count() > 0`?

Reply:
Right. But what's the technical difference between `Count() > 0` and `Any()` in this case?

Reply:
Technically the function Any checks if the list is greater than zero and not null.